whelktool/scripts/cleanups/2022/01/datakbse-556-sucia-uris/datakbse-556-suecia-uris.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+Update links to data.kb.se in associatedMedia.uri for Suecia docs
+
+- Replace .tif links with .tif links
+- Replace .zip links with data.kb.se landing page links 
+- Remove .pdf links (dataset doesn't exist anymore)
+- Also verify .tif file sizes in public note (they are all correct) 
+
+See DATAKBSE-556 for more information
+
+filesize.txt generated like this: 
+cat new-tifs.txt | while read URI; do echo $URI $(curl -s -IXGET $URI | grep 'Content-Length' | grep -o [0-9]*); done > filesize.txt
+
+ */
+PDF_LINK_PREFIX = 'https://data.kb.se/datasets/2016/05/suecia/'
+
+newTifs = getReportWriter("new-tifs.txt")
+sizeDiff = getReportWriter("size-diff.txt")
+
+Set<String> records = []
+Map<String,String> replacementLinks = [:]
+for (String line: new File(scriptDir, 'new_data.kb.se.tsv').readLines()) {
+    def (recordId, oldLink, newUri) = line.split('\t')
+    records.add(recordId)
+    String newPkgUri = newUri.split('/')[0..-2].join('/') // drop last component 
+    
+    if (oldLink.endsWith('.tif')) {
+        String filename = oldLink.split('/')[-1]
+        String newLink = newPkgUri + '/' + filename
+        newTifs.println(newLink)
+        replacementLinks.put(oldLink, newLink)
+    } else if (oldLink.endsWith('.zip')) {
+        replacementLinks.put(oldLink, newPkgUri)
+    }
+}
+
+Map<String, Integer> sizes = [:]
+for (String line: new File(scriptDir, 'filesizes.txt').readLines()) {
+    def (newUri, filesizeBytes) = line.split(' ')
+    int fileSizeMb = Math.round(Integer.parseInt(filesizeBytes) / (1024 * 1024))
+    sizes.put(newUri, fileSizeMb)
+}
+
+records.each { fix(it, replacementLinks, sizes) }
+
+void fix(String id, Map<String,String> replacementLinks, Map<String, Integer> sizes) {
+    if (id.isNumber()) {
```

Comment:
To be pedantic, it's the length that counts (`id.size() > 14`), since XL-ID:s *can* be all digits too.  (It is millisecs since 1970 in base 30, padded with a 7 character checksum). It got to length 15 (8+7) in 1972, 16 in 1992 and will be of length 17 in about 600 years. We didn't start counting from 0 (see the now removed https://github.com/libris/librisxl/commit/b40835709933f75cc38fef626e21fd7a0839a53a#diff-6c99ded845f7cd4c5e1bb17b4ef49e026460d9b1690da75c31d5445b58587238 (we may want to reinstate that for clarity?)).)

Not a problem with the given ID:s here of course. But there are already a whopping *two* all-digit XL-id:s in the system:  049160554683877 and 049118751660179. 🙈 (Appear to be broken items.)